File: TheBigDay/Models/Event.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TheBigDay.Models
{
    [Table("Event")]
    public class Event
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }
        public required DateTime StartDateTime { get; set; }
        public required DateTime EndDateTime { get; set; }
        public bool IsPrivate { get; set; }
        public bool GuestListFinalised { get; set; }
        public string? TicketLink { get; set; }
        public required string AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public required string Suburb { get; set; }
        public required string State { get; set; }
        public required string Country { get; set; }
        public required string Postcode { get; set; }
        public required EventType EventType { get; set; }
        public bool IsDeleted { get; set; }

        //dependancies 
        public List<EventItem>? EventItems { get; set; }
        public List<EventCustomer>? EventCustomer { get; set; }
    }
}

File: TheBigDay.Web/projects/common/src/lib/common-rest-models/event.ts
import {FormEntry} from "./form-entry";
import {Customer} from "./customer";

export interface Event {
    id?: string
    name: string;
    description: string;
    startDateTime: Date;
    endDateTime: Date;
    isPrivate: boolean;
    guestListFinalised: boolean;
    ticketLink?: string;
    addressLine1: string;
    addressLine2?: string;
    suburb: string;
    state: string;
    country: string;
    postcode: string;
    isDeleted: boolean;
    eventType: EventType

    // navigation
    eventItems?: EventItem[];
    eventCustomers?: EventCustomer[];
}

export interface EventType {
    id: string;
    name: string;
}

export interface EventCustomer {
    id: string;
    customerId: string;
    eventId: string;
    isAdmin: boolean;
    isCreator: boolean;

    // navigation
    customer: Customer;
    event: Event;
}

export interface EventItem {
    id: string;
    eventId: string;
    formEntryId: string;
    isFinalisedByCustomer: boolean;
    isFinalisedByStore: boolean;
    finalPriceByStore: number;

    // navigation
    event: Event;
    formEntry: FormEntry;
}

export const defaultEvent: Event = {
    addressLine1: "",
    country: "",
    description: "",
    endDateTime: new Date(),
    eventCustomers: [],
    eventItems: [],
    eventType: {
        id: '',
        name: ''
    },
    guestListFinalised: false,
    isDeleted: false,
    isPrivate: false,
    name: "",
    postcode: "",
    startDateTime: new Date(),
    state: "",
    suburb: "",
}
